### PR TITLE
reorg

### PR DESCRIPTION
**Pull request by Skhmt**
tl;dr: cleaned up the folder structure so 99% of users have a better experience

Overall idea is to move developer tools to tools/, move the main code to aidungeon/, and clean up some files. 

- moved the main code to aidungeon/
- moved convert_gp2_model.py to tools/
- moved datasets.md to tools/
- moved test-models.py to tools/
- moved data/ to tools/
- removed model.torrent as it's mostly used as a magnet now
- removed download_model.sh as the focus is moving to gpt-neo, but open to re-adding it if it's more generalized for other models
- removed changelog.txt as it hasn't been updated in nearly a year and a half
- added section to readme.md to show the full magnet links incase tinyurl.com dies
- created a launch.py in the main folder - both play scripts call it and the readme says to run that if you want to run manually - all it does is run the aidungeon module and it's named differently so it won't be confused for one of the play scripts
- created aidungeon/\_\_init\_\_.py and moved some purely display logic and the outermost game loop there from play.py
- modified imports so they work
- incorporated #161 

Should the install scripts and requirements text files be in their own folder too? 

This PR **_needs_** discussion before merging, but if merged right now, it would run.

### CODE REVIEW

**Comment by MikkoMMM** [comment]
This whole reorganizing business made me wonder whether it's even worth it to keep those tools in the same repository. Even though I hear about people finetuning stuff, I still doubt it's nearly a majority. It's hardly even a large file if we compare to the torch with CUDA, but that text_adventures.txt is still 29.7 MiB most people won't ever use, I'd expect.

**Comment by Skhmt** [comment]
> This whole reorganizing business made me wonder whether it's even worth it to keep those tools in the same repository. Even though I hear about people finetuning stuff, I still doubt it's nearly a majority. It's hardly even a large file if we compare to the torch with CUDA, but that text_adventures.txt is still 29.7 MiB most people won't ever use, I'd expect.

I was thinking about that too; that one text file is like 99% of the entire repo size.

I'm not sure why the GPT-Neo magnet is so much smaller, but I've been seeding it for a while and it seems to be working.

**Comment by MikkoMMM** [comment]
I tried it for a bit but it didn't find peers. Not until I reopened the magnet with those trackers attached to its end. It's neither here or there but I feel like it was a stupid idea to distribute those models compressed. It barely saves any data, and now if somebody wants to seed they have to keep the large compressed files in addition to the uncompressed ones. It makes the disincentive to do that that much higher. Maybe that's just me, though.

**Comment by Skhmt** [comment]
> It's neither here or there but I feel like it was a stupid idea to distribute those models compressed. It barely saves any data, and now if somebody wants to seed they have to keep the large compressed files in addition to the uncompressed ones. It makes the disincentive to do that that much higher. Maybe that's just me, though.

I completely agree. Tars literally just make life difficult in the context of torrents.

> I don't think the install scripts should be moved from the root directory. If people don't read the instructions they'll have a harder time finding them. And if you make it so they'd have to be called for example like "sh install/install.sh", it might not be intuitive without the instructions. The requirements files aren't as important to have in the root directory, but then you'd have yet another directory only to move 3 files out of sight. Maybe not a whole lot of benefit but some more text to the manual installing method (though it might not see as much use as the script methods).

Those are good points. I'll move the requirements files to a folder - I'd rather have one more folder and 3 fewer loose files, as less-technical users are less likely to dig through folders and they can be hidden away for the most part. 

We need the play scripts, launch.py, readme (for github to use it as a home page basically), license (for legal reasons), and config to be in the root for sure. I agree with keeping install there as even if they just double-click install.bat, it'll actually work fine.

**Comment by Meanmelter** [comment]
I would say let's also remove the training data but it seems it was restored at one point?
https://github.com/cloveranon/Clover-Edition/commit/3b8968bd23abda5ce906cfa18bc02b77738b8520

Anyways I like the new layout now, if we wanted to separate the Linux/Windows versions we should utilize GitHub Releases and put them in separate .ZIP files. I can test the Linux versions so shouldn't be an issue. That would remove the extra .bat or .sh file. Maybe we even wanna look into `py2exe`?

**Comment by Skhmt** [comment]
I've been messing with py2exe and can't get it to work, but I'm probably just not using it right

**Comment by Meanmelter** [comment]
Also, I did notice the play.bat file opens a new WT instance for me - Not a big deal though it gets rid of any profile settings you configure (I had a WT profile that would insta-launch clover w/background image)

**Comment by Meanmelter** [comment]
> 
> 
> Also, I did notice the play.bat file opens a new WT instance for me - Not a big deal though it gets rid of any profile settings you configure (I had a WT profile that would insta-launch clover w/background image)

NVM I found where I can just change the profile name 😅

**Comment by cloveranon** [comment]
People frequently want to look at the training data (e.g. to find out why the model constantly talks about Count Grey or "svelks".) Not to mention people that want to fine-tune. I would like it to be hosted somewhere and linked to, if it's not part of the code.

**Comment by Skhmt** [comment]
> People frequently want to look at the training data (e.g. to find out why the model constantly talks about Count Grey or "svelks".) Not to mention people that want to fine-tune. I would like it to be hosted somewhere and linked to, if it's not part of the code.

Ideally you'd set up a second repo to host `data/`, `datasets.md`, `convert_gpt2_model.py`, and `test-models.py` and we can link to it in the readme.
